### ISSUE TITLE
Benefit of ConcurrentDictionary lost if you don't use valueFactory version.

### ISSUE THREAD

**Issue by jsuddsjr** [zeppaman/Owin-Nancy-Tutorial#1]
At [EmbeddedStaticContentConventionBuilder.cs#L63](https://github.com/zeppaman/Owin-Nancy-Tutorial/blob/master/TestOwinNancy.Core/EmbeddedStaticContentConventionBuilder.cs#L63), you attempt to cache the returned `Func<Response>` object in a `ConcurrentDictionary`, but you are invoking the helper method _every time_ before calling the `GetOrAdd` function. 

If you were to replace this direct call to `BuildContentDelegate` with a `Func<TKey, TValue>` call, then `GetOrAdd` will not invoke the method _unless the key is not found in the dictionary_. See MSDN for [ConcurrentDictionary<TKey, TValue>.GetOrAdd Method (TKey, Func<TKey, TValue>)](https://msdn.microsoft.com/en-us/library/ee378677(v=vs.110).aspx).

**Comment by zeppaman** [comment]
You are right. I agree with you, it is a mistake. Thanks for reporting this, I'm going to fix it.
